client/client_test.go

```diff
@@ -95,17 +95,17 @@ func generateRedisProbeDef(name string) *probes_configpb.ProbeDef {
 
 // TestRegisterAndAddProbe tests that an extension probe type can be registered and added to Cloudprober without error.
 func TestRegisterAndAddProbe(t *testing.T) {
-	ctx, client := setupCloudproberAndClient(t)
-	defer teardownCloudproberAndClient(ctx, t, client)
+	client := setupCloudproberAndClient(t)
+	defer teardownCloudproberAndClient(t, client)
 
 	for i := 0; i < testRunCount; i++ {
 		probeName := fmt.Sprintf("testExtension%d", i)
-		if err := client.RegisterAndAddProbe(ctx, int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(probeName), &myprobe.Probe{}); err != nil {
+		if err := client.RegisterAndAddProbe(context.Background(), int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(probeName), &myprobe.Probe{}); err != nil {
```

Comment:
Just create a context.Background at the start, rather than constantly repeating it. Here and elsewhere.

Reply:
I changed it to calling context.Background() as per Balazs last comment:
    "No need for ctx here. Just use context.Background() everywhere."

Would you recommend `ctx := context.Background()` as per the previous version?

Reply:
Sorry, I should have stated it better: there's no reason to return a Context from the setup function, as context.Background() is to be used in tests by default (unless something else is required).
It's OK to declare a `ctx` variable and set it to context.Backround() if there are many calls in the test that expect a context. It's also OK to just use context.Background(). Your choice.

Reply:
Ah okay, I misunderstood, apologies. I've amended this to use a single `ctx` variable per test.